.github/workflows/docs.yml

```diff
@@ -0,0 +1,81 @@
+name: Deploy Docs to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "docs/**"
+      - ".github/workflows/docs.yml"
+
+  pull_request:
+    paths:
+      - "docs/**"
+      - ".github/workflows/docs.yml"
+
+  workflow_dispatch: {}
+
+permissions:
```

Comment:
It's better to put permissions on jobs, to avoid jobs that run user code to not have elevated privileges. Could you please move this to the `deploy` job, please? Thanks.